clock.go

```diff
@@ -71,29 +64,36 @@ func (clock *Clock) Now() time.Time {
 
 // After is part of the clock.Clock interface.
 func (clock *Clock) After(d time.Duration) <-chan time.Time {
-	defer clock.notifyAlarm()
-	clock.mu.Lock()
-	defer clock.mu.Unlock()
-	notify := make(chan time.Time, 1)
-	if d <= 0 {
-		notify <- clock.now
-	} else {
-		clock.setAlarm(clock.now.Add(d), func() { notify <- clock.now })
-	}
-	return notify
+	return clock.NewTimer(d).Chan()
+}
+
+func (clock *Clock) NewTimer(d time.Duration) clock.Timer {
+	c := make(chan time.Time, 1)
+	return clock.addAlarm(d, c, func() {
+		c <- clock.now
+	})
 }
 
 // AfterFunc is part of the clock.Clock interface.
 func (clock *Clock) AfterFunc(d time.Duration, f func()) clock.Timer {
+	return clock.addAlarm(d, nil, func() {
+		go f()
```

Comment:
could/should we use tomb to manage the lifetime of this goroutine? and the stop method should try to cleanup running routines as well.. could we have a test case for this scenario?